sirepo/template/jspec.py

```diff
@@ -396,6 +425,41 @@ def _generate_parameters_file(data):
     return template_common.render_jinja(SIM_TYPE, v)
 
 
+def _get_time_step_warning(run_dir):
+    def _get_rate(rates, i, j):
+        return abs(float(rates[i][1][j]))
+
+    def _get_max_rate(rates):
+        m =  _get_rate(rates, 0, 0)
+        for i in range(2):
+            t = rates[i][0]
+            assert 'IBS rate' in t or 'Electron cooling rate' in t, \
+                'unknown rates={}'.format(rates)
+            for j in range(3):
+                m = max(m, _get_rate(rates, i, j))
+        return m
+
+    m = _get_max_rate(get_rates(run_dir).rate)
+    w = None
+    r = 0.05 / m
+    s = simulation_db.read_json(
+        run_dir.join(template_common.INPUT_BASE_NAME),
+    ).models.simulationSettings.time_step
+    if s < 0.25 * r:
+        w = (
+            'The time step is too small. This can lead to long run times.'
+            ' Please consider decreasing the total time and/or increasing'
+            ' the number of steps.'
+        )
+    elif s > 4 * r:
+        w = (
+            'The time step is too large. This can lead to innacurate results.'
+            ' Please consider increasing the total time and/or decreasing'
+            ' the number of steps.'
+        )
```

Comment:
I'd recommend formatting the error message with newlines, in the same spot as the text breaks because the UI shows the message in a PRE formatted box.

>            'The time step is too small. This can lead to long run times.\n'
>            'Please consider decreasing the total time and/or increasing\n'
>            'the number of steps.'